src/mlpack/core/distributions/laplace_distribution_impl.hpp

```diff
@@ -49,7 +53,33 @@ inline void LaplaceDistribution::Probability(const arma::mat& x,
  *
  * @param observations List of observations.
  */
-inline void LaplaceDistribution::Estimate(const arma::mat& observations)
+template<typename MatType>
+[[deprecated("Will be removed in mlpack 5.0.0; use Train() instead")]]
+inline void LaplaceDistribution<MatType>::Estimate(const MatType& observations)
+{
+  Train(observations);
+}
+
+/**
+ * Estimate the Laplace distribution directly from the given observations,
+ * taking into account the probability of each observation actually being from
+ * this distribution.
+ */
+template<typename MatType>
+[[deprecated("Will be removed in mlpack 5.0.0; use Train() instead")]]
+inline void LaplaceDistribution<MatType>::Estimate(const MatType& observations,
+                                                   const VecType& probabilities)
+{
+  Train(observations, probabilities);
+}
```

Comment:
Nice and indeed, does not make sense and better for API compatibility to have Train function everywhere :+1: